types/codemirror/index.d.ts

```diff
@@ -1079,6 +1088,15 @@ declare namespace CodeMirror {
         that might change its proper indentation (only works if the mode supports indentation). Default is true. */
         electricChars?: boolean;
 
+        /** A regular expression used to determine which characters should be replaced by a special placeholder. Mostly useful for non-printing special characters. The default is /[\u0000-\u001f\u007f-\u009f\u00ad\u061c\u200b-\u200f\u2028\u2029\ufeff\ufff9-\ufffc]/. */
+        specialChars?: RegExp;
+
+        /** A function that, given a special character identified by the specialChars option, produces a DOM node that is used to represent the character. By default, a red dot (•) is shown, with a title tooltip to indicate the character code. */
+        specialCharPlaceholder?: (char: string) => HTMLElement;
```

Comment:
Should this be `Element` instead of `HTMLElement`?

Reply:
Good question. I'm not entirely sure, but you are _probably_ right. I'll go change it.

Reply:
Hm. Looks like `react-codemirror` disagrees?

Reply:
Where are you seeing that?

Reply:
The build failed, with the log mentioning `codemirror-react`. I'm not actually sure where to find that again (I already pushed a revert commit).

Reply:
Something weird is going on with CI not related to your changes. I think the change was fine. But we can merge it either way.

Reply:
Will keep it like this for now – maybe my first PR in this project is not the best one to fight CI over a change I'm not sure about. Thanks for your help!